sources/niLang/src/DataTablePath.cpp

```diff
@@ -240,11 +251,19 @@ tBool __stdcall cDataTablePath::Evaluate(iDataTable* apDT)
             return eFalse;
           }
           pCur = pCur->GetChildFromIndex(nS);
-          ++it;
-          continue;
+          break;
         }
-      case eDataTablePathOp_Property:
-        {
+        case eDataTablePathOp_ChildDataTable: {
+          const tU32 nS = pOp->GetValue().ULong();
+          if (nS >= pCur->GetNumChildren()) {
+#pragma niTodo("Better error report mech, for now it bloats the output, so its disabled.")
+            // niError(niFmt(_A("Can't find child at index '%s' (%d)."), pOp->GetValue(), nS));
```

Comment:
Idk, I agree in general, but for Expressions I just dump all the errors even though it's noisy. But here its more common to expect to not find the path...

Without any error message debugging the expressions is a bit of a lost cause xD

An error code (with the niCC throw_error) would be nicer but it wouldn't include an actual description of the error either.

Reply:
Makes sense. For the moment I just copied how the other branches were handling error reporting, we can change this later oncec we decide to cleanup various API's across niLang.